docs/source/user/input_file_overview.rst

```diff
@@ -0,0 +1,140 @@
+and which lines contained the value and key pair.  If a keyname is not found the
+parser will return an error with information about which line it was reading
+from.
+
+
+Modules using Key + Value Format
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following modules use the *key + value* format input files (all other
+modules use the *value column* format):
+
+============== ==========================================================
+ Module         Input file           
+============== ==========================================================
+AeroDyn         Main AD15 input file 
+AeroDyn         Airfoil files
+InflowWind      Main IfW input file
+InflowWind      Uniform wind input file
+InflowWind      Bladed wind summary file
+ServoDyn        Main ServoDyn input file
+ServoDyn        Structural control submodule input file
+ServoDyn        Structural control sumbodule prescribed force input file
+SubDyn          SubDyn SSI matrix input files
+============== ==========================================================
+
+Reasons for change
+~~~~~~~~~~~~~~~~~~
+
+The main reason for the change in the input file parsing was to allow for the
+passing of a complete input file in memory from a wrapper code into OpenFAST or
+a module.  For example, when including the AeroDyn module into a Python code,
+the input file can be passed in directly in memory without writing to disk
+first.  This helps reduce the IO overhead in optimization loops where the module
+might be called many times sequentially with very small changes to the input
+file.  *NOTE: this is still a work in progress, so not all modules can be linked
+this way yet*.
+
+To accomplish this, the file parser written by Marshall Buhl for parsing airfoil
+tables in AeroDyn 15 in FAST8 was used.  This parser included the more robust
+*key + value* input format.
+
+
+
+.. _sec_troubleshoot_input_file:
+
+Troubleshooting input files
+---------------------------
+
+When troubleshooting an input file error, try the following procedure:
+
+1. An error message containing a line number and variable name, the file format
+   being parsed is a *key + value* format.  Check that the key is spelled
+   exactly as the input file.  See :numref:`sec_format_key_value` below.
+2. An error message containing only the variable name but no line number is a
+   *value column* input file format.  See :numref:`sec_value_column` below.
+3. Turn on `echo` option in the input file and check the resulting `.ech` for
+   which line the file parsing stopped at. This may help isolate where the input
+   file parsing failed when no line number is given in the error message.
+4. Compare the problematic input file with an input file of the same type from
+   the regression test suite distributed with OpenFAST.  See section
+   :numref:`testing` for details on the regression tests, or check the
+   repository at `r-test <https://github.com/openfast/r-test>`__ .
+
+
+..
+   Input file type by module
```

Comment:
I'm not sure what this table is for.

Reply:
This table is commented out and won't render.  I started adding it, but decided to just leave it as a commented out block for later (if we want it).